app/managers/performance_metrics_manager/index.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, type AppStateStatus} from 'react-native';
+import performance from 'react-native-performance';
+
+import Batcher from './performance_metrics_batcher';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'mobile_channel_switch' |
+    'mobile_team_switch';
+
+class PerformanceMetricsManager {
+    private target: Target;
+    private batchers: {[serverUrl: string]: Batcher} = {};
+    private hasRegisteredLoad = false;
+    private lastAppStateIsActive = AppState.currentState === 'active';
+
+    constructor() {
+        AppState.addEventListener('change', (appState) => this.onAppStateChange(appState));
+    }
+
+    private onAppStateChange(appState: AppStateStatus) {
+        const isAppStateActive = appState === 'active';
```

Comment:
Similar to my comments on your PR about the websocket, I don't think we actually need to store the previous value of this if this is only called when the app state changes

Reply:
There are several "non active states", and this is executed when the state become non active. Calling several times `forceSend` shouldn't have any effect, but still, there is no need to do so. So I think there is not a big issue to store the previous value, to only call this the first time we go to an "inactive state".